src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -610,6 +610,67 @@ mongoc_uri_parse_auth_mechanism_properties (mongoc_uri_t *uri, const char *str)
    return true;
 }
 
+
+static bool
+mongoc_uri_check_srv_service_name (mongoc_uri_t *uri, const char *str)
+{
+   /* 63 character DNS query limit, excluding prepended underscore. */
+   const size_t mongoc_srv_service_name_max = 62;
+
+   size_t length = 0u;
+   size_t num_alpha = 0u;
+   size_t i = 0u;
+   char prev = '\0';
+
+   BSON_ASSERT_PARAM (uri);
+   BSON_ASSERT_PARAM (str);
+
+   length = strlen (str);
+
+   /* Initial DNS Seedlist Discovery Spec: This option specifies a valid SRV
+    * service name according to RFC 6335, with the exception that it may exceed
+    * 15 characters as long as the 63rd (62nd with prepended underscore)
+    * character DNS query limit is not surpassed. */
+   if (length > mongoc_srv_service_name_max) {
+      return false;
+   }
+
+   /* RFC 6335: MUST be at least 1 character. */
+   if (length == 0) {
```

Comment:
Also noticed you're using `0` here but `0u` in another comparison below.

Reply:
Good spot. Will update/fix.